src/test/java/hudson/security/LDAPEmbeddedTest.java

```diff
@@ -612,7 +614,10 @@ public void userValidityAttributes() throws Exception {
         LDAPSecurityRealm realm = new LDAPSecurityRealm(Collections.singletonList(configuration), false, null, null, null);
         r.jenkins.setSecurityRealm(realm);
         r.configRoundtrip();
-        assertFalse(realm.loadUserByUsername2("amy").isEnabled());
-        assertFalse(realm.loadUserByUsername2("bender").isAccountNonExpired());
+        assertThrows(DisabledException.class, () -> realm.loadUserByUsername2("amy"));
+        assertThrows(DisabledException.class, () -> User.getById("amy", true).impersonate2());
+        assertThrows(AccountExpiredException.class, () -> realm.loadUserByUsername2("bender"));
+        assertThrows(AccountExpiredException.class, () -> User.getById("bender", true).impersonate2());
+        assertThrows(FailingHttpStatusCodeException.class, () -> r.createWebClient().withBasicApiToken("amy").goTo(""));
```

Comment:
Currently a 500, with the `DisabledException` being printed to the system log, pending some nicer handler along the lines of `AccessDeniedHandlerImpl`.

Reply:
Would that be a core change? Or can I specify an `AccessDeniedHandler` bean in this plugin?

Reply:
Core change I guess. Not sure it is worth the effort anyway.